components/copy_button.tsx

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState} from 'react';
+import {FormattedMessage} from 'react-intl';
+import {Tooltip} from 'react-bootstrap';
+
+import OverlayTrigger from 'components/overlay_trigger';
+
+import Constants from 'utils/constants';
+import {copyToClipboard} from 'utils/utils.jsx';
+
+type Props = {
+    content: string;
+    beforeCopyText?: string;
+    afterCopyText?: string;
+    idMessage?: string;
+};
+
+const CopyButton: React.FC<Props> = (props: Props) => {
+    const [isCopied, setIsCopied] = useState(false);
+
+    const copyText = (e: React.MouseEvent<HTMLAnchorElement, MouseEvent>): void => {
+        e.preventDefault();
+        setIsCopied(true);
+
+        setTimeout(() => {
+            setIsCopied(false);
+        }, 5000);
+
+        copyToClipboard(props.content);
+    };
+
+    const tooltip = (
+        <Tooltip id='copyButton'>
+            <FormattedMessage
+                id={props.idMessage}
```

Comment:
The `id` for these messages is for translations, so we need two different messages IDs for this. You can look at other uses of `FormattedMessage` or at `i18n/en.json` for how those look.

We also need to make sure that the translation strings are captured by our translation tools which will make them appear in `i18n/en.json` when those are run. If you pass a constant as the `id` prop of a `FormattedMessage`, they'll get picked up automatically, but if you can't do that, you can wrap them in the `t` function from `utils/i18n` like `t('some.id')`.

Reply:
@hmhealey is this what you meant regarding the ids?

Also, any idea what to do about the failing tests on index.test.js? They basically fail because the renderer now returns the div block for the node processor.

Reply:
That's almost right. I'll follow up with a couple comments which will fix the failing `i18n-check` CI step.

As for the tests, we could have the tests for the renderer code just check for the div with the extra information attached, and then we could add a test for `messageHtmlToComponent` that confirms that it can convert the special div to a `CodeBlock` component. Breaking up the testing into multiple steps should work well enough now that the logic is spread across a few components